features/discovery/src/main/java/org/opennms/netmgt/discovery/messages/DiscoveryJob.java

```diff
@@ -98,4 +101,25 @@ public String toString() {
                  .add("ranges", m_ranges)
                  .toString();
     }
+    
+    /**
+     * <P>
+     * Returns int : the total task timeout for all ip ranges
+     * </P>
+     */
+    public int calculateTaskTimeout(){
+    	BigInteger taskTimeOut = new BigInteger("0");
+        for(final IPPollRange range : m_ranges) {
+        	BigInteger retries = new BigInteger(Integer.toString(range.getRetries()));
+        	BigInteger sizeOfIpAddrRange = range.getAddressRange().getSizeOfIpAddrRange(); 
+        	BigInteger timeOut = new BigInteger(Long.toString(range.getTimeout()));    
+        	
+        	BigInteger temp = ((retries.add(new BigInteger("1"))).multiply(sizeOfIpAddrRange.multiply(timeOut)));
+        	BigDecimal bd = new BigDecimal(temp);
+        	
+        	taskTimeOut = taskTimeOut.add(bd.multiply(BigDecimal.valueOf(IPAddrRange.FUDGE_FACTOR)).toBigInteger());
+        }
+        return taskTimeOut.intValue();
```

Comment:
If the value is greater than Integer.MAX_VALUE, you need to return Integer.MAX_VALUE instead of using intValue().